### PR TITLE
CSP-1410: GetEmployerRelationships Endpoint

### PR DESCRIPTION
**Pull request by shaun-downey-education**
- Added **EmployerRelationshipsController** with a **GetEmployerRelationships** endpoint to allow the access of employer relationships by HashedId, Ukprn and account legal entity PublicHasedId.

### CODE REVIEW

**Review by PeterDavidWilliams — DISMISSED** [review]

File: src/SFA.DAS.PR.Api/Controllers/EmployerRelationshipsController.cs [hunk 1]
```diff
@@ -17,9 +17,10 @@ public class EmployerRelationshipsController(IMediator _mediator) : ActionRespon
     [HttpGet("employeraccount/{accountHashedId}")]
     [Authorize(Policy = Policies.Management)]
     [ProducesResponseType(typeof(GetEmployerRelationshipsQueryResult), StatusCodes.Status200OK)]
-    public async Task<IActionResult> GetEmployerRelationships(string accountHashedId, CancellationToken cancellationToken)
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
```
Comment (DevenKShah) [inline src/SFA.DAS.PR.Api/Controllers/EmployerRelationshipsController.cs:20]: Is this meant to be 400 Bad request?

File: src/SFA.DAS.PR.Application/EmployerRelationships/Queries/GetEmployerRelationships/GetEmployerRelationshipsQueryResult.cs [hunk 2]
```diff
@@ -6,10 +6,10 @@ public GetEmployerRelationshipsQueryResult()
 
     }
 
-    public GetEmployerRelationshipsQueryResult(List<AccountLegalEntityPermissionsModel> accountLegalEntities)
+    public GetEmployerRelationshipsQueryResult(List<AccountLegalEntityPermissionsModel>? accountLegalEntities)
```
Comment (DevenKShah) [inline src/SFA.DAS.PR.Application/EmployerRelationships/Queries/GetEmployerRelationships/GetEmployerRelationshipsQueryResult.cs:9]: You do not need default ctor as you have made this nullable

**Review by DevenKShah — DISMISSED** [review]
Few questions I have, but this looks good.

File: src/SFA.DAS.PR.Data/Repositories/EmployerRelationshipsReadRepository.cs [hunk 3]
```diff
@@ -6,16 +6,33 @@ namespace SFA.DAS.PR.Data.Repositories;
 
 public class EmployerRelationshipsReadRepository(IProviderRelationshipsDataContext providerRelationshipsDataContext) : IEmployerRelationshipsReadRepository
 {
-    public async Task<Account?> GetRelationships(string accountHashedId, CancellationToken cancellationToken)
+    public async Task<Account?> GetRelationships(string accountHashedId, long? ukprn, string? accountlegalentityPublicHashedId, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (shaun-downey-education) [reply src/SFA.DAS.PR.Data/Repositories/EmployerRelationshipsReadRepository.cs:9]: @DevenKShah We've validated the account via the account HashedId but the other two properties are not mandatory when requesting permissions.

**Comment by DevenKShah** [comment]
Closing this as 1465 made these changes redundant.

**Pull request closed without merging**